File: 0div-idbig-omega-company-tagsdiv-idbig-omega-topbardiv-classcompanytagscontainer-styleoverflow-x-scroll-flex-wrap-nowrap-div-classcompanytagscontainer-tagno-companies-found-for-this-problem-div-divdiv-classcompanytagscontainer-chevrondivsvg-version11-idicon-xmlnshttp-wwww3org-2000-svg-xmlns-xlinkhttp-wwww3org-1999-xlink-x0px-y0px-viewbox0-0-32-32-fill4087f1-xml-spacepreserve-stylewidth-20px-darkreader-inline-fill-4aa0f2-data-darkreader-inline-fillpolygon-points16-22-6-12-74-106-16-192-246-106-26-12-polygonrect-id-x3c-transparent-rectangle-x3e-classst0-fillnone-width32-height32-rect-svg-div-div-div-div2824-count-pairs-whose-sum-is-less-than-target/0div-idbig-omega-company-tagsdiv-idbig-omega-topbardiv-classcompanytagscontainer-styleoverflow-x-scroll-flex-wrap-nowrap-div-classcompanytagscontainer-tagno-companies-found-for-this-problem-div-divdiv-classcompanytagscontainer-chevrondivsvg-version11-idicon-xmlnshttp-wwww3org-2000-svg-xmlns-xlinkhttp-wwww3org-1999-xlink-x0px-y0px-viewbox0-0-32-32-fill4087f1-xml-spacepreserve-stylewidth-20px-darkreader-inline-fill-4aa0f2-data-darkreader-inline-fillpolygon-points16-22-6-12-74-106-16-192-246-106-26-12-polygonrect-id-x3c-transparent-rectangle-x3e-classst0-fillnone-width32-height32-rect-svg-div-div-div-div2824-count-pairs-whose-sum-is-less-than-target.java
// import java.util.List;
// import java.util.Collections;
// import java.util.Arrays;

class Solution {

    public int countPairs(List<Integer> integerList, int target) {

        // integerList = integerList.toArray(new Integer[0]);
        
        Collections.sort(integerList);
        // Arrays.sort(integerList);
        
        int numOfPairs = 0;
        int leftIndex = 0;
        // int backIndex = integerList.length - 1;
        int backIndex = integerList.size() - 1;

        while (leftIndex < backIndex) {
            
            if (integerList.get(leftIndex) + integerList.get(backIndex) < target) {
                int currNumOfPairs = backIndex - leftIndex;
                numOfPairs += currNumOfPairs;
                leftIndex++;
            } 
            
            else {
                backIndex--;
            }
        }
        
        return numOfPairs;
    }
}

File: 0div-idbig-omega-company-tagsdiv-idbig-omega-topbardiv-classcompanytagscontainer-styleoverflow-x-scroll-flex-wrap-nowrap-div-classcompanytagscontainer-tagno-companies-found-for-this-problem-div-divdiv-classcompanytagscontainer-chevrondivsvg-version11-idicon-xmlnshttp-wwww3org-2000-svg-xmlns-xlinkhttp-wwww3org-1999-xlink-x0px-y0px-viewbox0-0-32-32-fill4087f1-xml-spacepreserve-stylewidth-20px-darkreader-inline-fill-4aa0f2-data-darkreader-inline-fillpolygon-points16-22-6-12-74-106-16-192-246-106-26-12-polygonrect-id-x3c-transparent-rectangle-x3e-classst0-fillnone-width32-height32-rect-svg-div-div-div-div2824-count-pairs-whose-sum-is-less-than-target/0div-idbig-omega-company-tagsdiv-idbig-omega-topbardiv-classcompanytagscontainer-styleoverflow-x-scroll-flex-wrap-nowrap-div-classcompanytagscontainer-tagno-companies-found-for-this-problem-div-divdiv-classcompanytagscontainer-chevrondivsvg-version11-idicon-xmlnshttp-wwww3org-2000-svg-xmlns-xlinkhttp-wwww3org-1999-xlink-x0px-y0px-viewbox0-0-32-32-fill4087f1-xml-spacepreserve-stylewidth-20px-darkreader-inline-fill-4aa0f2-data-darkreader-inline-fillpolygon-points16-22-6-12-74-106-16-192-246-106-26-12-polygonrect-id-x3c-transparent-rectangle-x3e-classst0-fillnone-width32-height32-rect-svg-div-div-div-div2824-count-pairs-whose-sum-is-less-than-target.cpp
// #include <iostream>
// #include <vector>
// #include <algorithm>

class Solution {
    
    public:
    int countPairs(std::vector<int> array, int target) {

        std::sort(array.begin(), array.end());
        
        int numOfPairs = 0;
        int leftIndex = 0;
        int backIndex = array.size() - 1;

        while (leftIndex < backIndex) {
            
            if (array[leftIndex] + array[backIndex] < target) {

                int currNumOfPairs = backIndex - leftIndex;
                
                numOfPairs += currNumOfPairs;
                leftIndex++;
        } 
            else {
                backIndex--;
            }
        }
        return numOfPairs;
    }
    int main() {
        std::vector<int> array = {-6, 2, 5, -2, -7, -1, 3};
        int target = -2;
        std::cout << countPairs(array, target) << std::endl;
        return 0;
    };
};